### DISCUSSION TITLE
Current positions

### DISCUSSION THREAD

**Discussion by Kitosight** [microsoft/vscode-discussions#414] [Extension Development QnA] [answered]
How can I get the current character & line position of the cursor?

**Comment by ArturoDent** [comment, marked as answer]
Look at the `selection` property of the textEditor: and then `active.character` and `active.line`.

**Reply by Kitosight** [reply]
Alright, thanks.
I haven't tested it yet, but I'd like to note the reason why I was looking for it.

I was making Auto Complion Items, but encountered the issue that it doesn't match the `$`. I was told to use `Range` to solve the issue. I partially solved it, but encountered another issue - I actually don't know how to use it, so completion items were provided only on the first line of the first character.

Therefore, I thought about getting the current line & character number in order to do something with it, but I guess I'll do something wrong again, so it won't match or insert completion items correctly.

Can you help with this?
The current `Range` code is
```js
const range = new vscode.Range(new vscode.Position(0,0), new vscode.Position(0,1));
```
> If you still remember what you just read, it provides only on the first line of the first character because I did this `Range` wrongly.
